rotkehlchen/chain/evm/decoding/gearbox/gearbox_cache.py

```diff
@@ -0,0 +1,327 @@
+                        address=pool.underlying_token,
+                        token_kind=EvmTokenKind.ERC20,
+                        weight=ONE,
+                    )],
+                    encounter=TokenEncounterInfo(
+                        description='Querying gearbox pools',
+                        should_notify=False,
+                    ),
+                )
+                verified_farming_tokens.add(token.identifier)
+            except NotERC20Conformant as e:
+                log.error(
+                    f'Skipping pool {pool} because {token_address} is not a '
+                    f'valid ERC20 token. {e}',
+                )
+                continue
+
+        # ensure the lp tokens exist in the db
+        verified_lp_tokens: set[str] = set()
+        for token_address in pool.lp_token_addresses:
+            try:
+                token = get_or_create_evm_token(
+                    userdb=evm_inquirer.database,
+                    evm_address=string_to_evm_address(token_address),
+                    chain_id=evm_inquirer.chain_id,
+                    evm_inquirer=evm_inquirer,
+                    underlying_tokens=[UnderlyingToken(
+                        address=pool.underlying_token,
+                        token_kind=EvmTokenKind.ERC20,
+                        weight=ONE,
+                    )],
+                    encounter=TokenEncounterInfo(
+                        description='Querying gearbox pools',
+                        should_notify=False,
+                    ),
+                )
+                verified_lp_tokens.add(token.identifier)
+            except NotERC20Conformant as e:
+                log.error(
+                    f'Skipping pool {pool} because {token_address} is not a '
+                    f'valid ERC20 token. {e}',
+                )
+                continue
+
+        pool.farming_pool_addresses = verified_farming_tokens
+        pool.lp_token_addresses = verified_lp_tokens
+        verified_pools.append(pool)
+    return verified_pools
+
+
+def query_gearbox_data_from_chain(
+        evm_inquirer: 'EvmNodeInquirer',
+        existing_pools: set[ChecksumEvmAddress],
+) -> list[GearboxPoolData] | None:
+    """
+    Query all Gearbox information(lp tokens, pools, lp coins) from data compressor.
+
+    May raise:
+    - RemoteError if failed to query chain
+    """
+    data_compressor = evm_inquirer.contracts.contract(GEARBOX_DATA_COMPRESSOR)
+    try:
+        pools_data = data_compressor.call(
+            node_inquirer=evm_inquirer,
+            method_name='getPoolsV3List',
+        )
+    except DeserializationError as e:
+        log.error(f'Gearbox data compressor returned an invalid pool address. {e}')
+        return None
+
+    gearbox_farming_pool_abi = evm_inquirer.contracts.abi('GEARBOX_FARMING_POOL')
+    new_pools = []
+    for pool_data in pools_data:
+        pool_address = deserialize_evm_address(pool_data[0])
```

Comment:
it shouldn't but this can raise deserializationError. Catch it, log and continue